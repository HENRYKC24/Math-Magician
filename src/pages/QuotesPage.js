import React, { useState, useEffect } from 'react';

const Quotes = () => {
  const [currentQuote, setCurrentQuote] = useState({
    dataAvailable: false,
    quote: '',
    author: '',
  });

  const getQuote = () => {
    setCurrentQuote(() => ({
      dataAvailable: false,
    }));

    fetch('https://random-math-quote-api.herokuapp.com/')
      .then((result) => result.json())
      .then((data) => setCurrentQuote({
        ...data,
        dataAvailable: true,
      }));
  };

  useEffect(() => getQuote(), []);
  const { dataAvailable, quote, author } = currentQuote;

  return (
    <div className="quotes">
      {!dataAvailable && <p>Loading...</p>}
      <p className="quote">
        {quote && `${quote}`}
        <br />
        <span className="quote">{quote && `- ${author}`}</span>
      </p>
      {quote && <button className="change-quote" type="button" onClick={getQuote}>Change Quote</button>}
    </div>
  );
};

export default Quotes;
