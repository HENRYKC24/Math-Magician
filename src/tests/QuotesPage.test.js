import React from 'react';
import renderer from 'react-test-renderer';
import QuotesPage from '../pages/QuotesPage';

it('Quotes page renders the UI as expected', () => {
  const tree = renderer.create(<QuotesPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
