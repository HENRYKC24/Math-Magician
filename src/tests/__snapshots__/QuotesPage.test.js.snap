// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes page renders the UI as expected 1`] = `
<div
  className="quotes"
>
  <p
    className="loading"
  >
    Loading...
  </p>
  <p
    className="quote"
  >
    
    <br />
    <span
      className="quote"
    >
      
    </span>
  </p>
  
</div>
`;
